activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -9,7 +9,8 @@ module AttributeMethods
     include ActiveModel::AttributeMethods
 
     included do
-      initialize_generated_modules
+      @attribute_methods_builder = nil
```

Comment:
This is a smell that that memoization is happening too early. Now this implementation has to know about an internal instance variable in the `AM` module.